wordpress-mobile/WordPress-Android
# Quick Start for Existing Users V2 - Changes to Existing Tasks - Reader (Part ii)

Parent #16349

This PR makes following main changes for the `FOLLOW_SITE` task on the `Reader` tab:

1. Moves existing QS logic from reader post list to parent reader fragment and vm in 6b6264e (no. of lines have increased due to this). 
    #### [This is done as `Discover` tab fragment `ReaderDiscoverFragment`  is separate to other tab fragments (which are instances of `ReaderPostListFragment`). Both `DiscoverFragment`  and `ReaderPostListFragment` are created from `ReaderFragment` and it is easier to manage QS steps from this fragment + vm when we introduce `Discover` tab step which involves observing tab changes in the `ReaderViewModel`.]
2. Displays QS focus point for `Settings` menu. Note that previously QS focus point was not shown for `Search` menu. The task is marked complete when `Settings` menu is tapped.

3. ~~Adds intermediate `Discover` tab step if not on `Discover` tab.~~ Auto-selects`Discover` tab step if not on `Discover` tab.

https://user-images.githubusercontent.com/1405144/165953529-0977953b-0be2-4cde-9938-1079c97eceaa.mp4

To test:

1. Launch the app with a site having a quick start in progress.
2. Go to `My Site` tab.
3. Select `Grow your audience` from `Next Steps` card on `My Site` screen. The card will be present on the `Initial Screen `tab set in the `Settings`.
5. Click `Connect with other sites` row.
6. ✅ Notice that quick start flow for the task is updated as follows:
    <img width=400 src ="https://user-images.githubusercontent.com/1405144/165957395-9e65c0f4-9995-4354-b047-d6874b188a3a.png"/>
   Discover tab is auto-selected if not already selected.
7. ✅ Notice that quick start focus points are removed on navigating away from the `Reader` tab but retained on device rotation.
8. ✅ Notice that the task is marked complete only when `Settings` menu step of Reader Discover is completed. This can be checked by enabling `Collect information` in `Me` -> `App Settings `-> `Privacy settings` and checking for log:
 `🔵 Tracked: quick_start_task_completed, Properties: {"default_tab_experiment":<tab name>,"task_name":"follow_site"}`
or by running unit test with description: `given pending follow site qs task, when settings menu clicked, then setting menu step is completed`

Review Instructions:

Review tests collectively at the end (as opposed to commit-by-commit) as there were changes to the flow.

## Regression Notes
1. Potential unintended areas of impact
- Quick start focus points, snackbars are not updated based on whether the user was already on the Discover tab or not.
- Any crashes due to adding/ updating custom views for `Discover` tab and `Settings` menu.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually (see To Test: section) and added unit tests.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests in `ReaderViewModel` for updated quick start steps.

PR submission checklist:
